privilege/privileges/cache.go

```diff
@@ -909,7 +909,7 @@ func (p *MySQLPrivilege) RequestVerification(activeRoles []*auth.RoleIdentity, u
 // DBIsVisible checks whether the user can see the db.
 func (p *MySQLPrivilege) DBIsVisible(user, host, db string) bool {
 	if record := p.matchUser(user, host); record != nil {
-		if record.Privileges != 0 {
+		if record.Privileges&mysql.GlobalDBVisible > 0 {
```

Comment:
Actually we can define `GlobalDBVisible` in the TiDB repo if it's not already added to the parser repo.